tests/mir-opt/pre-codegen/slice_iter.forward_loop.PreCodegen.after.panic-unwind.mir

```diff
@@ -120,10 +100,8 @@ fn forward_loop(_1: &[T], _2: impl Fn(&T)) -> () {
         StorageLive(_9);
         _9 = _4 as *mut T (PtrToPtr);
         StorageLive(_10);
-        StorageLive(_22);
         _10 = _9 as *const T (PointerCoercion(MutToConstPointer));
         _11 = NonNull::<T> { pointer: _10 };
-        StorageDead(_22);
```

Comment:
And for this kind of thing specifically, note that the old one was saying "_22 is live", not using it, then saying "_22 is dead", which was always just a bug.  No longer having the unnecessary storage annotations here is amazing.